lib/sumo/client.rb

```diff
@@ -79,8 +110,12 @@ def creds
   end
   private :creds
 
-  def connection
-    @connection ||= Excon.new('https://api.sumologic.com')
+  def connection(endpoint = nil)
```

Comment:
I was thinking that theoretically on redirects we cannot predict the endpoint, that’s why I thought about just creating a new connection (as it is in the original redirect follower middleware). But sure, in practice we will usually have like a handful of hostnames and no more, and storing them makes sense.